app/models/manageiq/providers/openstack/infra_manager/event_parser.rb

```diff
@@ -1,43 +1,14 @@
 module ManageIQ::Providers::Openstack::InfraManager::EventParser
-  def self.event_to_hash(event, ems_id)
-    content = message_content(event, ems_id)
-    event_type = content["event_type"]
-    payload = content["payload"] || {}
-
-    log_header = "ems_id: [#{ems_id}] " unless ems_id.nil?
-    _log.debug("#{log_header}event: [#{event_type}]") if $log && $log.debug?
-
-    # attributes that are common to all notifications
-    event_hash = {
-      :event_type => event_type,
-      :source     => "OPENSTACK",
-      :message    => payload,
-      :timestamp  => content["timestamp"],
-      :username   => content["_context_user_name"],
-      :full_data  => event,
-      :ems_id     => ems_id
-    }
-
-    if payload.key? "instance_id"
-      event_hash[:host_id] = Host.find_by(:uid_ems => payload["instance_id"]).try(:id)
-      event_hash[:host_name] = payload["instance_id"]
-    end
-    event_hash[:message]                   = payload["message"]           if payload.key? "message"
-    event_hash[:host_ems_ref]              = payload["node"]              if payload.key? "node"
-    event_hash[:availability_zone_ems_ref] = payload["availability_zone"] if payload.key? "availability_zone"
-    event_hash[:chain_id]                  = payload["reservation_id"]    if payload.key? "reservation_id"
-    event_hash
-  end
-
-  def self.message_content(event, ems_id)
-    unless ems_id.nil?
-      ems = ExtManagementSystem.find_by_id(ems_id)
-      if ems.connection_configuration_by_role("amqp")
-        if event[:content].key?("oslo.message")
-          return JSON.parse(event[:content]["oslo.message"])
-        end
+ def self.event_to_hash(event, ems_id)
+    ManageIQ::Providers::Openstack::EventParserCommon.event_to_hash(event, ems_id) do |event_hash, payload|
+      if payload.key? "instance_id"
+        event_hash[:host_id] = Host.find_by(:uid_ems => payload["instance_id"]).try(:id)
```

Comment:
Not related to this change but we should setting the `event_hash[:host_uid_ems]` and then doing this lookup in the event handler worker like we do with the [vm uid_ems](https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_event.rb#L80)